### PR TITLE
Add remaining Waypoint install view sections

### PR DESCRIPTION
**Pull request by ashleemboyer**
## "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The description template has been filled in below

---

## Relevant links

- [Preview link](https://dev-portal-git-ambadd-remaining-install-view-sections-hashicorp.vercel.app/waypoint/downloads) 🔎

## What/Why

- Adds `OfficialReleasesSection` and `FeaturedTutorialsSection` subcomponents local to `ProductDownloadsView`. The `FeaturedTutorialsSection` may become reusable in the future across many pages, but I think we can move it to the main components folder when necessary. 
- Adds data for the "Featured Tutorials" section in `src/data/waypoint-install.json`
- Adds a `getTruncatedText` helper function under `src/lib/get-truncated-text.ts`, used for truncating the Learn Tutorial descriptions before rendering them in the "Featured Tutorials" section

## How

Nothing too interesting here. 😊 

## Testing

- [ ] Go to [/waypoint/downloads on the preview link](https://dev-portal-git-ambadd-remaining-install-view-sections-hashicorp.vercel.app/waypoint/downloads)
- [ ] Make sure the Official Releases section behaves as you expect
- [ ] Make sure the Featured Tutorials section behaves as you expect

## Anything else?

There's something wrong with the hover state of the card under Official Releases and I'm not sure what yet. But I think it's under the scope of the CardLink component, so my vote is to fix it separately from this PR unless it's a super quick fix. I tried to debug the issue but haven't had success yet.

### CODE REVIEW

File: src/views/product-downloads-view/components/featured-tutorials-section/featured-tutorials-section.module.css [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  color: var(--token-color-neutral-foreground-primary);
+  margin-bottom: 12px;
+}
+
+/*
+TODO: these @media queries come from the numbers defined in src/pages/style.css
+but I wasn't able to get the @media --mobile and @media --tablet queries to work
+like they do in that file. The code below will eventually need to be refactored,
+this is a temporary copy/paste to get the new install view out.
+*/
```
Comment (BRKalow) [inline src/views/product-downloads-view/components/featured-tutorials-section/featured-tutorials-section.module.css:11]: Interesting, so the custom media queries aren't working then? Maybe worth adding a ticket to track 👀

File: src/views/product-downloads-view/components/featured-tutorials-section/index.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+              <div>
+                <Text className={s.cardTitle} size={300} weight="semibold">
+                  {title}
+                </Text>
+                <Text className={s.cardBody} size={200} weight="regular">
+                  {getTruncatedText(description, 80)}
+                </Text>
+              </div>
+              <div className={s.cardIcons}>
+                <IconPlay16 />
```
Comment (BRKalow) [inline src/views/product-downloads-view/components/featured-tutorials-section/index.tsx:45]: I think the icons here denote whether or not the tutorial has a video or interactive example, maybe we can add a TODO to add support for that?
Screen Shot 2022-02-25 at 10 52 26 AM

**Review by BRKalow — APPROVED** [review]
👍

File: src/views/product-downloads-view/components/featured-tutorials-section/featured-tutorials-section.module.css (hunk 1 above)
Comment (ashleemboyer) [reply src/views/product-downloads-view/components/featured-tutorials-section/featured-tutorials-section.module.css:11]: They weren't working in this spt, so I'm not sure if they're setup incorrectly or what's happening. They are still working for the hide-on-(tablet/mobile) classes that they affect.

File: src/views/product-downloads-view/components/featured-tutorials-section/index.tsx (hunk 2 above)
Comment (ashleemboyer) [reply src/views/product-downloads-view/components/featured-tutorials-section/index.tsx:45]: I'm pretty sure these will eventually be `LearnTutorialCard` components, so I'll add a TODO referencing that Asana ticket. 😊

**Review by BRKalow — APPROVED** [review]

**Pull request merged**